bugsnag-android-core/src/main/java/com/bugsnag/android/Session.java

```diff
@@ -89,15 +128,56 @@ void setAutoCaptured(boolean autoCaptured) {
         this.autoCaptured.set(autoCaptured);
     }
 
+    /**
+     * Determines whether a cached session payload is v1 (where only the session is stored)
+     * or v2 (where the whole payload including app/device is stored).
+     *
+     * @return whether the payload is v2
+     */
+    boolean isV2Payload() {
+        return file != null && file.getName().endsWith("_v2.json");
+    }
+
     @Override
     public void toStream(@NonNull JsonStream writer) throws IOException {
-        writer.beginObject()
-            .name("id").value(id)
-            .name("startedAt").value(DateUtils.toIso8601(startedAt));
-
-        if (user != null) {
-            writer.name("user").value(user);
+        if (file != null) {
+            if (isV2Payload()) {
+                serializeV2Payload(writer);
+            } else {
+                serializeV1Payload(writer);
+            }
+        } else {
+            writer.beginObject();
+            writer.name("notifier").value(Notifier.INSTANCE);
+            writer.name("app").value(app);
+            writer.name("device").value(device);
+            writer.name("sessions").beginArray();
+            serializeSessionInfo(writer);
+            writer.endArray();
+            writer.endObject();
         }
+    }
+
+    private void serializeV2Payload(@NonNull JsonStream writer) throws IOException {
+        writer.value(file);
+    }
+
+    private void serializeV1Payload(@NonNull JsonStream writer) throws IOException { // TODO test me
```

Comment:
See TODO comment :)

Reply:
This is addressed by existing test coverage in `SessionV1PayloadTest` - I just forgot to remove this comment.